tests/ui/default_mismatches_new.stderr

```diff
@@ -0,0 +1,75 @@
+error: consider delegating to the auto-derived `Default` for `DeriveIntDefault`
+  --> tests/ui/default_mismatches_new.rs:112:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         Self { value: 0 }
+   | ||_________________________- help: use: `Self::default()`
+LL | |      }
+   | |______^
+   |
+   = note: `-D clippy::default-mismatches-new` implied by `-D warnings`
+   = help: to override `-D warnings` add `#[allow(clippy::default_mismatches_new)]`
+
+error: consider delegating to the auto-derived `Default` for `DeriveTupleDefault`
+  --> tests/ui/default_mismatches_new.rs:121:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         Self(0)
+   | ||_______________- help: use: `Self::default()`
+LL | |      }
+   | |______^
+
+error: consider delegating to the auto-derived `Default` for `NonZeroDeriveDefault`
+  --> tests/ui/default_mismatches_new.rs:130:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         Self(42)
+   | ||________________- help: use: `Self::default()`
+LL | |      }
+   | |______^
+
+error: consider delegating to the auto-derived `Default` for `ExtraBlockDefault`
+  --> tests/ui/default_mismatches_new.rs:139:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         { Self::default() }
+   | ||___________________________- help: use: `Self::default()`
+LL | |      }
+   | |______^
+
+error: consider delegating to the auto-derived `Default` for `ExtraBlockRetDefault`
+  --> tests/ui/default_mismatches_new.rs:148:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         return {
+...  ||
+LL | ||         };
+   | ||__________- help: use: `Self::default()`
+LL | |      }
+   | |______^
+
+error: consider delegating to the auto-derived `Default` for `MultiStatements`
+  --> tests/ui/default_mismatches_new.rs:163:22
+   |
+LL |        fn new() -> Self {
+   |  _______________________^
+LL | |/
+LL | ||         println!("Hello, world!");
+LL | ||         let i = 42;
+LL | ||         Self(i)
+   | ||_______________- help: use: `Self::default()`
+LL | |      }
+   | |______^
```

Comment:
Here is an example of an unhelpful error message: it doesn't tell what the problem is, it only gives a suggestion to "fix" it.